src/os/bluestore/BlueStore.cc

```diff
@@ -5058,7 +5057,7 @@ int BlueStore::_write_out_fm_meta(uint64_t target_size)
 
 int BlueStore::_open_alloc()
```

Comment:
@ifed01 How about splitting `_open_alloc` into `_create_alloc` and `_open_alloc` ?
Then it could be done that first pass on `BlueFS::mount()` could do all necessary `init_rm_free`, not second one.
Somehow I like it better, but being honest, I do not have any heavy arguments for that. Minor would be that we could get rid of `bluefs_shared_alloc_context_t::set` interface, and shared_alloc will always be set for lifetime of BlueFS.

Reply:
@aclamk - tried this, seems not viable since we can't proceed with allocator init (i.e. init_rm_free calls) on the first mount - one needs to read free extents' list from DB first.